pkg/fake/ec2api.go

```diff
@@ -38,14 +38,16 @@ import (
 )
 
 type CapacityPool struct {
-	CapacityType string
-	InstanceType string
-	Zone         string
+	CapacityType  string
+	InstanceType  string
+	Zone          string
+	ReservationID string
 }
 
 // EC2Behavior must be reset between tests otherwise tests will
 // pollute each other.
 type EC2Behavior struct {
+	DescribeCapacityReservationsOutput  AtomicPtr[ec2.DescribeCapacityReservationsOutput]
```

Comment:
nit: You could consider using `MockedFunction` and you get some of the scaffolding for free

Reply:
Good point, I pretty much just copied the reset of the DescribeCalls. Maybe a good first issue to migrate all of them?

Reply:
Sure, wanna open it?

Reply:
Discussed offline: Migrating to `MockedFunction` is more complicated because we are sort-of combining setting explicit outputs and filtering those outputs -- which sort of breaks from the general testing pattern that we were trying to establish with `MockedFunction` -- we agreed that we can take this in a follow-on that we'll track in the same issue